app/models/reservation.rb

```diff
@@ -95,9 +95,11 @@ def self.in_range(start_time, end_time)
   def self.upcoming(t = Time.zone.now)
     # If this is a named scope differences emerge between Oracle & MySQL on #reserve_end_at querying.
     # Eliminate by letting Rails filter by #reserve_end_at
-    reservations = find(:all, conditions: "reservations.canceled_at IS NULL AND (orders.state = 'purchased' OR orders.state IS NULL)", order: "reserve_end_at asc", joins: ["LEFT JOIN order_details ON order_details.id = reservations.order_detail_id", "LEFT JOIN orders ON orders.id = order_details.order_id"])
-    reservations.delete_if { |r| r.reserve_end_at < t }
-    reservations
+    joins("LEFT JOIN order_details ON order_details.id = reservations.order_detail_id")
+      .joins("LEFT JOIN orders ON orders.id = order_details.order_id")
+      .where(canceled_at: nil, "orders.state" => [nil, "purchased"])
+      .order(reserve_end_at: :asc)
+      .delete_if { |reservation| reservation.reserve_end_at < t }
```

Comment:
I'd like to. I didn't dig up a lot in history about why it's necessary I've left it alone for now.